File: demo/cpp/tcp_pair.cpp
#include "cc/neolux/utils/unitransmit/Unitransmit.h"

#include <chrono>
#include <cstdint>
#include <iostream>
#include <string>
#include <thread>
#include <vector>

using cc::neolux::utils::unitransmit::UniTransmit;

int main() {
    UniTransmit server("tcp-s://0.0.0.0:9100");
    UniTransmit client("tcp://127.0.0.1:9100");

    client.write(std::string("hello"));
    std::this_thread::sleep_for(std::chrono::milliseconds(50));

    auto data = server.read_all();
    std::string out(reinterpret_cast<const char *>(data.data()), data.size());
    std::cout << "recv: " << out << "\n";

    return 0;
}

File: demo/python/tcp_pair.py
from __future__ import annotations

import time

from unitransmit import UniTransmit


def main() -> None:
    server = UniTransmit("tcp-s://0.0.0.0:9100")
    client = UniTransmit("tcp://127.0.0.1:9100")

    client.write(b"hello")
    time.sleep(0.1)

    print("recv:", server.read_all())


if __name__ == "__main__":
    main()
